expo/expo
# feature(cli): add vscode compatibility handler to inspector proxy

# Why

Builds upon #21449

This adds a handler that makes sure the `vscode-js-debug` extension can be used with the custom inspector proxy. These changes should NOT interfere with the other debuggers.

# How

- Added vscode compatibility handler
  - Add missing `description` to function nodes, as required by vscode-js-debug
    _(before https://github.com/microsoft/vscode-js-debug/issues/1583)_
  - Respond to `Debugger.getPossibleBreakpoints`, Hermes doesn't respond to this message
- ~~Filter out "useless" `callFrames` when debugging~~ 
  **Moved back to PR, we need to limit filtering**
  → https://github.com/expo/expo/pull/21663
- Avoid Hermes binding breakpoints to wrong locations, before sourcemap is loaded 
  → https://github.com/expo/expo/pull/21631
- Avoid Hermes crashing by disabling `Runtime.getProperties` on `symbol` types
  → https://github.com/expo/expo/pull/21639

# Test Plan

- See tests
- ~~Test project with vscode, see [this repository](https://github.com/byCedric/vscode-js-debug-issue-1583) for a test project.~~
- See https://github.com/expo/expo/pull/21641 to test `apps/bare-expo`

# Checklist
